frontend/dockerfile/dockerfile_lint_test.go

```diff
@@ -418,6 +419,34 @@ COPY Dockerfile .
 	})
 }
 
+func testUndefinedArg(t *testing.T, sb integration.Sandbox) {
```

Comment:
Add checks for no warnings for base image variables (eg. things defined in alpine base image) and `ENV/ARG` defined in parent stage for multi-stage builds.